### PR TITLE
Add dependency graph viewer tool

### PR DESCRIPTION
**Pull request by davidwrighton**
- This is a minimally functional tool used to view the dependency graphs produced by crossgen2
- This was previously present in the CoreRT repo, but did not make the transition to this repo when the type system was moved

Instructions for use
- Launch it, and then run a compilation using crossgen2
- It will build an in memory graph of every dependency node that has been added to the main graph. This graph is live and is updated as the compiler runs, and so can be used as a debugging aid
- Double click on the graph in the first list box, this will open up a filter view which can be filled in with a regex to filter the set of graphs seen, or just the empty string. (If using the empty string, press the filter button).
- Then individual nodes can be examined. Double click on them to show the related nodes to the specific node.

Also I've added the WhyDgml tool, which allows finding the reason why a specific node is present and produces a complete dependency chain.
-  It works best if FirstMarkLogStrategy is used. It might hit cycles if full marking strategy is used, so better not try that. That's untested.
- Given the name of the DGML file and a name of the node in the DGML graph, it prints the path from the node to the roots.

### CODE REVIEW

**Comment by davidwrighton** [comment]
@dotnet/crossgen-contrib  I've added the existing dependency analysis viewer here. Does anyone feel a need to polish it further, or should we find some other place to put this. I feel that its time to stop storing this code in the CoreRT repo, but I'm not sure the runtime repo is the right place for this.

**Comment by jkotas** [comment]
I think it is fine to store the viewer tool in this repo. I am not sure about the `src\coreclr\tools\ILCompiler-DependencyGraph-Viewer` directory name. It is a mouthful. What about `src\coreclr\tools\aot\DependencyGraphViewer`? 

Also, I think it would be a good idea to copy over https://github.com/dotnet/corert/tree/master/src/ILCompiler.DependencyAnalysisFramework/WhyDgml tool as well. I have used both DependencyGraphViewer and WhyDgml myself. Each of these is used for different type of troubleshooting.

File: src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/ILCompiler-DependencyGraph-Viewer.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <RootNamespace>DependencyLogViewer</RootNamespace>
+    <AssemblyName>ILCompiler-DependencyGraph-Viewer</AssemblyName>
+
+    <ApplicationManifest>app.manifest</ApplicationManifest>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Diagnostics.Tracing.TraceEvent">
+      <Version>$(TraceEventVersion)</Version>
+    </PackageReference>  </ItemGroup>
```
Comment (am11) [inline src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/ILCompiler-DependencyGraph-Viewer.csproj:16]: nit, indentation is a bit off:
```suggestion
  <ItemGroup>
    <PackageReference Include="Microsoft.Diagnostics.Tracing.TraceEvent" Version="$(TraceEventVersion)" />
  </ItemGroup>
```

File: src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/App.config [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/App.config:1]: This file is probably not needed now that this is a .NET core app.

**Comment by mangod9** [comment]
@davidwrighton is this good to merge?

**Comment by davidwrighton** [comment]
@mangod9, Not quite, I've just responded to the feedback, and it needs an approving code review

File: src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/ILCompiler-DependencyGraph-Viewer.csproj (hunk 1 above)
Comment (davidwrighton) [reply src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/ILCompiler-DependencyGraph-Viewer.csproj]: Thanks. I've fixed this.

File: src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/App.config (hunk 2 above)
Comment (davidwrighton) [reply src/coreclr/tools/ILCompiler-DependencyGraph-Viewer/App.config]: Looks to not be required. I've removed it.

File: src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphViewer.csproj [hunk 3]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCoreAppToolCurrent)-windows</TargetFramework>
+    <UseWindowsForms>true</UseWindowsForms>
+    <OutputType>Exe</OutputType>
+    <RootNamespace>DependencyLogViewer</RootNamespace>
+
+    <ApplicationManifest>app.manifest</ApplicationManifest>
```
Comment (jkotas) [inline src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphViewer.csproj:9]: app.manifest should not be required with .NET Core.

**Review by jkotas — APPROVED** [review]

File: src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphViewer.csproj (hunk 3 above)
Comment (MichalStrehovsky) [reply src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphViewer.csproj:9]: The tool needs to run elevated - without elevation it will just fail.

Comment (jkotas) [reply src/coreclr/tools/aot/DependencyGraphViewer/DependencyGraphViewer.csproj:9]: Ok, I have missed the elevation part.

**Pull request merged**